sympy/stats/tests/test_continuous_rv.py

```diff
@@ -231,6 +231,33 @@ def test_beta():
     assert expand_func(E(B)) == a / S(a + b)
     assert expand_func(variance(B)) == (a*b) / S((a + b)**2 * (a + b + 1))
 
+@XFAIL
```

Comment:
Why is this marked as `@XFAIL`?

Reply:
Because `dens(z) == (Sum(z**(_k + a - 1)*(c/2)**_k*(1 - z)**(b - 1)*exp(-c/2)/(beta(_k + a, b)*factorial(_k)), (_k, 0, oo)))` is giving `False`, when it is `True`. I searched for similar case and found `UniformSumDistribution` in `sympy/stats/crv_types.py` to have the same problem. They have also solved it using `@XFAIL`, otherwise their tests are also failing. So I have used `@XFAIL`

Reply:
What is LHS of the comparison returning?

Reply:
Same thing ie `(Sum(z**(_k + a - 1)*(c/2)**_k*(1 - z)**(b - 1)*exp(-c/2)/(beta(_k + a, b)*factorial(_k)), (_k, 0, oo)))`
I am getting the following error.
```
>>> density(X)(z)
(Sum(z**(_k + a - 1)*(c/2)**_k*(1 - z)**(b - 1)*exp(-c/2)/(beta(_k + a, b)*factorial(_k)), (_k, 0, oo)))
>>> density(X)(z) == (Sum(z**(_k + a - 1)*(c/2)**_k*(1 - z)**(b - 1)*exp(-c/2)/(beta(_k + a, b)*factorial(_k)), (_k, 0, oo)))
False
```

Reply:
That is because `_k` is a `Dummy` and `Dummy('x') == Dummy('x')` returns `False`.
You can use `str` for comparison here and in other places (and remove `@XFAIL`).
For example, `UniformSum` test could be modified to:
```
>>> str(density(X)(x)) == 'Sum((-1)**_k*(-_k + x)**(n - 1)*binomial(n, _k), (_k, 0, floor(x)))/factorial(n - 1)'
True
```

Reply:
Ok. I will also modify for `BetaNoncentral` and also `UniformSum`.

Reply:
@sidhantnagpal I have added the required tests. Please review.